### PR TITLE
Upgrade to Django 1.8

### PR DESCRIPTION
**Pull request by clintonb**
XCOM-441

FYI @rlucioni @jimabramson @Nickersoft this is purely discovery.

### CODE REVIEW

**Comment by rlucioni** [comment]
@clintonb [relevant](https://docs.djangoproject.com/en/1.8/releases/1.8/#django-conf-urls-patterns).

TL;DR: `urlpatterns` should now be a list of `django.conf.urls.url()` instances.

**Comment by clintonb** [comment]
Ready for review @rlucioni.

File: ecommerce/extensions/payment/tests/test_processors.py [hunk 1]
```diff
@@ -101,9 +100,7 @@ def test_get_transaction_parameters(self):
         """ Verify the processor returns the appropriate parameters required to complete a transaction. """
 
         # Patch the datetime object so that we can validate the signed_date_time field
-        with mock.patch.object(processors.cybersource.datetime, u'datetime',
-                               mock.Mock(wraps=datetime.datetime)) as mocked_datetime:
```
Comment (rlucioni) [inline ecommerce/extensions/payment/tests/test_processors.py:105]: Any specific reason to remove this? You're so against mocking `datetime`!

**Comment by rlucioni** [comment]
:shipit: @clintonb .

File: ecommerce/extensions/payment/tests/test_processors.py (hunk 1 above)
Comment (clintonb) [inline ecommerce/extensions/payment/tests/test_processors.py:105]: I honestly don't care whether we mock `datetime` or not. Doing so is simply inconvenient. In this case, your mock caused issues with Django's sqlite3 library. Specifically, the mocked value was not returning a proper type value. This was the simplest workaround.

Comment (rlucioni) [inline ecommerce/extensions/payment/tests/test_processors.py:105]: Cool, thanks for the clarification.

**Pull request merged**